linera-views/src/backends/dynamo_db.rs

```diff
@@ -926,21 +834,30 @@ impl ReadableKeyValueStore for DynamoDbStoreInternal {
     async fn find_keys_by_prefix(
         &self,
         key_prefix: &[u8],
-    ) -> Result<DynamoDbKeys, DynamoDbStoreInternalError> {
+    ) -> Result<Vec<Vec<u8>>, DynamoDbStoreInternalError> {
         let result_queries = self
             .get_list_responses(KEY_ATTRIBUTE, &self.start_key, key_prefix)
             .await?;
-        Ok(DynamoDbKeys { result_queries })
+        let mut keys = Vec::new();
+        for key in result_queries.iterate_keys() {
+            keys.push(key?.to_vec());
+        }
+        Ok(keys)
```

Comment:
Would probably allocate the right vec size right away (also below):

```suggestion
        result_queries.iterate_keys()
            .map(|key| key.map(|k| k.to_vec()))
            .collect()
```

Reply:
Piece of cake for claude